src/utils/blockedMonitor.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AzureFunctionsRpcMessages as rpc } from './../../azure-functions-language-worker-protobuf/src/rpc';
+import LogCategory = rpc.RpcLog.RpcLogCategory;
+import LogLevel = rpc.RpcLog.Level;
+import blockedAt = require('blocked-at');
+
+export function startBlockedMonitor(channel: { log: (log: rpc.IRpcLog) => void }, threshold = 100): NodeJS.Timer {
+    let blockedHistory: { time: string; duration: number; stack: string; resource: string }[] = [];
+
+    //threshold - minimum miliseconds of blockage to report.
+    //other parameters are default, more details on https://github.com/naugtur/blocked-at.
+    blockedAt(
+        (ms, stack, resource) => {
```

Comment:
What information is `resource` supposed to contain? When I test I just see `{"type":"TickObject"}` or `{"type":"PROMISE"}` which doesn't mean much to me.

Looking at the docs, maybe we have to set `resourcesCap` for it to be useful?

Reply:
Unfortunately the documentation does not explain the `resource` in details, if the signature provides the parameter why do not log it even it can be not very useful?

`resourceCap` was left as default as it needs experimenting for different scarious.